android/src/main/java/com/mapbox/mapboxgl/MapboxOfflineManager.java

```diff
@@ -0,0 +1,333 @@
+            notificationManager = context.getSystemService(NotificationManager.class);
+            notificationManager.createNotificationChannel(channel1);
+
+        }
+
+
+
+    }
+
+    public void downloadRegion(final String regionName) {
+        // Define offline region parameters, including bounds,
+        // min/max zoom, and metadata
+
+        // Start the progressBar
+//        startProgress();
+
+        // Create offline definition using the current
+        // style and boundaries of visible map area
+
+        map.getStyle(new Style.OnStyleLoaded() {
+            @Override
+            public void onStyleLoaded(@NonNull Style style) {
+                String styleUrl = style.getUri();
+                LatLngBounds bounds = map.getProjection().getVisibleRegion().latLngBounds;
+
+                double minZoom = map.getCameraPosition().zoom;
+                double maxZoom = map.getMaxZoomLevel();
+                float pixelRatio = context.getResources().getDisplayMetrics().density;
+                OfflineTilePyramidRegionDefinition definition = new OfflineTilePyramidRegionDefinition(
+                        styleUrl, bounds, minZoom, maxZoom, pixelRatio);
+
+                // Build a JSONObject using the user-defined offline region title,
+                // convert it into string, and use it to create a metadata variable.
+                // The metadata variable will later be passed to createOfflineRegion()
+                byte[] metadata;
+                try {
+                    JSONObject jsonObject = new JSONObject();
+                    jsonObject.put(JSON_FIELD_REGION_NAME, regionName);
+                    String json = jsonObject.toString();
+                    metadata = json.getBytes(JSON_CHARSET);
+                } catch (Exception exception) {
+
+                    Log.d(TAG,"Failed to encode metadata: "+ exception.getMessage());
+                    metadata = null;
+                }
+                // Create the offline region and launch the download
+                offlineManager.createOfflineRegion(definition, metadata, new OfflineManager.CreateOfflineRegionCallback() {
+                    @Override
+                    public void onCreate(OfflineRegion _offlineRegion) {
+                        Log.d( TAG,"Offline region created: " + regionName);
+                        offlineRegion = _offlineRegion;
+                        launchDownload(regionName);
+                    }
+                    @Override
+                    public void onError(String error) {
+                        Log.e( TAG, "Error: "+ error);
+                    }
+                });
+            }
+        });
+    }
+
+
+    private void launchDownload(String regName) {
+        // Set up an observer to handle download progress and
+        // notify the user when the region is finished downloading
+        final NotificationCompat.Builder downloadNotification = new NotificationCompat.Builder(context, CHANNEL_1_ID)
```

Comment:
I think it would be best to expose at least notification title into Flutter, so that users of the library have control over that. Example usage would be to have localized titles, for example

Reply:
Also, users of the library would probably want to have their own icon on the download notification